### PR TITLE
Typescript migration

### PR DESCRIPTION
**Pull request by cindywu**
This PR is a second attempt at https://github.com/jellypbc/poster/pull/372. 

- [x] `yarn add typescript`
- [x]  add `tsconfig.json`
- [x] update javascript packages
- [x] create `types.ts`
- [x] place types in types.ts `/app/javascript/components/types.ts`
- [x] get `yarn lint` to pass
- [x] add `@types/prosemirror-commands` `@types/prosemirror-model`
- [x] import types for existing libraries used including prosemirror
- [x] migrate all .js files in `/app/javascript/components` to .tsx (Editor.js left)
- [x] replace most any types in types.ts
- [ ] migrate Editor.js to Editor.tsx
- [ ] migrate all .js files in `/app/javascript/components/editor-config` to .ts

@dennyluan I think we should merge this and close off the typescript migration project. I could work on this for another month (or longer), but I don't think that is the best use of our resources. I'd like for us to move onto cleaning up our redux stuff.

I think the current PR gives enough of a blueprint for us to follow. 

@alexkrolick if you come across this and have thoughts or suggestions, would love to hear them!

My recommendation is we create `.tsx` files for new react components with the intention of no implicit anys and strict null checks. All javascript files should be `.ts` moving forward. For existing code, we fix and resolve as it conveniently overlaps with our product priorities.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +21,18 @@
     "@rails/ujs": "^6.0.0",
     "@rails/webpacker": "^5.2.1",
     "@reduxjs/toolkit": "^1.2.3",
+    "@types/prosemirror-commands": "^1.0.4",
```
Comment (dennyluan) [inline package.json:24]: Can these be moved to `devDepencies`?

**Review by dennyluan — APPROVED** [review]
Looks like great progress! WAhoo!

**Comment by dennyluan** [comment]
There's still a good number of any's in here, but I think it's a good start and something we can continue to work on 🤞

File: package.json (hunk 1 above)
Comment (cindywu) [reply package.json]: Done. I removed all prosemirror type dependencies and added @types/prosemirror-state to devDependencies.

**Pull request merged**